DNN Platform/DotNetNuke.Web.Client/Controls/ClientResourceExclude.cs

```diff
@@ -2,47 +2,53 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information
 
-namespace DotNetNuke.Web.Client.Controls
-{
-    using System;
-    using System.Web.UI;
+namespace DotNetNuke.Web.Client.Controls
+{
+    using System;
+    using System.Web.UI;
 
-    using ClientDependency.Core;
-    using ClientDependency.Core.Controls;
+    using DotNetNuke.Abstractions.ClientResources;
+    using DotNetNuke.Web.Client.Cdf;
 
-    public abstract class ClientResourceExclude : Control
-    {
-        public string Name { get; set; }
-
+    /// <summary>
+    /// Represents a control that excludes a client resource from being included in the page.
+    /// </summary>
+    public abstract class ClientResourceExclude : Control
+    {
+        private readonly IClientResourcesController clientResourcesController;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ClientResourceExclude"/> class.
+        /// </summary>
+        /// <param name="clientResourcesController">The client resources controller.</param>
+        protected ClientResourceExclude(IClientResourcesController clientResourcesController)
+        {
+            this.clientResourcesController = clientResourcesController;
+        }
+
+        /// <summary>
+        /// Gets or sets the name of the client resource to exclude.
+        /// </summary>
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Gets the dependency type of the client resource to exclude.
+        /// </summary>
         public ClientDependencyType DependencyType { get; internal set; }
 
-        /// <inheritdoc/>
-        protected override void OnPreRender(EventArgs e)
-        {
-            base.OnPreRender(e);
-            var loader = this.Page.FindControl("ClientResourceIncludes");
-            this.Name = this.Name.ToLowerInvariant();
-
-            if (loader != null)
-            {
-                ClientDependencyInclude ctlToRemove = null;
-                if (!string.IsNullOrEmpty(this.Name))
-                {
-                    foreach (ClientDependencyInclude ctl in loader.Controls)
-                    {
-                        if (ctl.Name.ToLowerInvariant() == this.Name && ctl.DependencyType == this.DependencyType)
-                        {
-                            ctlToRemove = ctl;
-                            break;
-                        }
-                    }
-                }
-
-                if (ctlToRemove != null)
-                {
-                    loader.Controls.Remove(ctlToRemove);
-                }
-            }
-        }
-    }
-}
+        protected override void OnInit(EventArgs e)

```

Comment:
By moving this process from OnPreRender which is executed after postbacks & event information has been processed to OnInit which is before all of that and very early, I believe we will be introducing a breaking change as inside of a PageLoad, etc you wouldn't be able to unload a resource, which you can do today.